[FIX] website: fix mega menu after scroll expending issue

dtda-odoo opened the pull request:
Steps to Reproduce: 
1. Add Megamenu to the website.
2. Switch to Edit mode.
3. Shift view to mobile view.
4. Scroll the view and open the navbar from the hamburger icon.
5. Try opening Megamenu.

Issue:
On mobile view, the mega menu’s sub-items weren’t appearing after scrolling. 
This was because the menu was using position: fixed, which removed it from the 
normal document flow and prevented it from expanding to fit its child elements 
properly. As a result, sub-menus were either cut off or not displayed at all.

Fix:
Changed the menu's positioning from fixed to absolute. Unlike fixed, absolute 
allows the element to remain positioned relative to its nearest-positioned 
ancestor. This keeps the menu within the layout flow of its container and lets 
it expands naturally based on its content, ensuring sub-items are fully visible 
and usable on scroll.

Task: 4684074

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/204439.png)](https://mergebot.odoo.com/odoo/odoo/pull/204439)

divy-odoo commented:
Hello @dtda-odoo, Here is the few review points to be consider. 

There few new bugs introduced after your changes,
![image](https://github.com/user-attachments/assets/c5c1665b-4a32-41e8-b289-53a2abd7dba9)

In commit message, 
1. Add the task-id
2. Introduce the clear step to reproduce the particular issue.

Thanks : )

dtda-odoo commented:
Hello @divy-odoo,
I have updated the previous fix to a new one. This aims to resolve the introduced bug and the bug specified in the task.
Also updated the commit message to have the following suggestions.
Can you please once again look into it?
Thanks.

divy-odoo commented on addons/website/static/src/scss/website.scss:
```diff
@@ -1660,7 +1660,7 @@ header {
     // Div needed for selector specificity to apply mh
     div.o_mega_menu {
-        position: fixed !important;
+        position: absolute;
```
Why we removed `!important`? Don't you think we should keep it?

divy-odoo reviewed (commented):
Thanks for the work, there are small comments to be resolve.

- Title issue in commit message doesn't mean here with steps to reproduce you can remove that. 
- Add task id at the bottom of commit message. ( refer other merged commit as well )

robodoo commented:
@dtda-odoo the target branch 'saas-18.1' has been disabled, you may want to close this PR.

divy-odoo commented:
Hello @dtda-odoo , You'll need to re-target the branch to saas-18.2. let me know once it's done. thanks : )

robodoo commented:
This PR targets the disabled branch odoo/odoo:saas-18.1, it needs to be retargeted before it can be merged.

msh-odoo commented:
@dtda-odoo The issue is not producible now

dtda-odoo commented:
The issue has been resolved: closing this PR.
Thanks
